reports/challenge groups/challenge_groups.qmd

```diff
@@ -0,0 +1,283 @@
+  "Multicard" = multicard_sold,
+  "Multifamily" = multifamily_sold,
+  "Top Decile" = top_decile_sold
+)
+
+
+gte_n <- \(n_sales, min_n, fn, na_type) {
+  ifelse(sum(!is.na(n_sales)) >= min_n, fn, na_type)
+}
+
+min_n <- 10
+
+rs_fns_list <- list(
+  cod_no_sop = \(x, y) gte_n(y, 2, cod(x / y, na.rm = TRUE), NA_real_),
+  prd_no_sop = \(x, y) gte_n(y, 2, prd(x, y, na.rm = TRUE), NA_real_),
+  prb_no_sop = \(x, y) gte_n(y, 2, prb(x, y, na.rm = TRUE), NA_real_),
+  mki_no_sop = \(x, y) gte_n(y, 2, mki(x, y, na.rm = TRUE), NA_real_),
+  cod = \(x, y) gte_n(y, min_n, cod(x / y, na.rm = TRUE), NA_real_),
+  cod_met = \(x, y) gte_n(y, min_n, cod_met(cod(x / y, na.rm = TRUE)), NA),
+  prd = \(x, y) gte_n(y, min_n, prd(x, y, na.rm = TRUE), NA_real_),
+  prd_met = \(x, y) gte_n(y, min_n, prd_met(prd(x, y, na.rm = TRUE)), NA),
+  prb = \(x, y) gte_n(y, min_n, prb(x, y, na.rm = TRUE), NA_real_),
+  prb_met = \(x, y) gte_n(y, min_n, prb_met(prb(x, y, na.rm = TRUE)), NA),
+  mki = \(x, y) gte_n(y, min_n, mki(x, y, na.rm = TRUE), NA_real_),
+  mki_met = \(x, y) gte_n(y, min_n, mki_met(mki(x, y, na.rm = TRUE)), NA),
+  min = \(x) min(x, na.rm = TRUE),
+  q25 = \(x) quantile(x, na.rm = TRUE, probs = 0.25),
+  median = \(x) median(x, na.rm = TRUE),
+  median_sale = \(x, y) median(y, na.rm = TRUE),
+  median_ratio = \(x, y) median(x / y, na.rm = TRUE),
+  q75 = \(x) quantile(x, na.rm = TRUE, probs = 0.75),
+  max = \(x) max(x, na.rm = TRUE)
+)
+
+## Some functions require 2 or 1 arguments.
+apply_functions <- function(dataset, dataset_name) {
+  x <- dataset$pred_pin_initial_fmv
+  y <- dataset$sale_ratio_study_price
+
+  results_rs <- lapply(rs_fns_list, function(f) {
+    if (length(formals(f)) == 2) {
+      f(x, y)
+    } else {
+      f(x)
+    }
+  })
+
+  results_df <- as.data.frame(results_rs)
+  results_df$count <- nrow(dataset)
+  results_df$median_prior_near_tot <- median(dataset$prior_near_tot, na.rm = TRUE)
+  results_df$dataset <- dataset_name
+
+  return(results_df)
+}
+
+results <- bind_rows(
+  lapply(names(datasets_sold), function(dataset_name) {
+    apply_functions(datasets_sold[[dataset_name]], dataset_name)
+  })
+)
+
+results %>%
+  as.data.frame() %>%
+  mutate(
+    median_change = ((median - median_prior_near_tot) / median_prior_near_tot) * 100,
+    min = round(min),
+    q25 = round(q25),
+    median = round(median),
+    `median previous estimate` = round(median_prior_near_tot),
+    q75 = round(q75),
+    max = round(max)
+  ) %>%
+  select(dataset, count, "median yoy change" = median_change, "median ratio" = median_ratio, cod, prb, prd, mki, min, q25, "median predicted" = median, "median sale" = median_sale, "median previous estimate", q75, max) %>%
+  kable(format = "html", digits = 2, row.names = FALSE, format.args = list(big.mark = ","), align = rep("l", 15)) %>%
+  kable_styling()
+```
+
+```{r _challenge_groups_plot_function, results = 'asis'}
+## Interactive elements are hashed out in order to produce rendered version
```

Comment:
**suggestion:** IMO it's incredibly useful to make these into plotly plots so we can see individual PINs. See the overall performance report for an example.